model/task_queue_service_dependency.go

```diff
@@ -227,16 +213,35 @@ func (d *basicCachedDAGDispatcherImpl) rebuild(items []TaskQueueItem) error {
 
 	sorted, err := topo.SortStabilized(d.graph, nil)
 	if err != nil {
-		grip.Alert(message.WrapError(err, message.Fields{
-			"dispatcher":                 DAGDispatcher,
-			"function":                   "rebuild",
-			"message":                    "problem ordering the tasks and associated dependencies within the DirectedGraph",
-			"distro_id":                  d.distroID,
-			"initial_num_taskqueueitems": len(items),
-			"num_task_groups":            len(d.taskGroups),
-		}))
-
-		return errors.Wrap(err, "topologically sorting the dependency graph")
+		unorderableNodes, ok := err.(topo.Unorderable)
+		if !ok {
+			grip.Alert(message.WrapError(err, message.Fields{
+				"dispatcher":                 DAGDispatcher,
+				"function":                   "rebuild",
+				"message":                    "problem ordering the tasks and associated dependencies within the DirectedGraph",
+				"distro_id":                  d.distroID,
+				"initial_num_taskqueueitems": len(items),
+				"num_task_groups":            len(d.taskGroups),
+			}))
+
+			return errors.Wrap(err, "topologically sorting the dependency graph")
+		}
+
+		var cycles [][]string
```

Comment:
Should this be `make([][]string, 0, len(unorderableNodes))` for consistency with L232?

Reply:
done